metadata-io/src/test/java/com/linkedin/metadata/graph/neo4j/Neo4jGraphServiceTest.java

```diff
@@ -153,4 +162,45 @@ public void testConcurrentRemoveNodes() {
     throw new SkipException("Neo4jGraphService produces duplicates");
   }
 
+  @Test
+  public void testPopulatedGraphServiceGetLineageMultihop() throws Exception {
+    GraphService service = getLineagePopulatedGraphService();
+
+    EntityLineageResult upstreamLineage = service.getLineage(datasetOneUrn, LineageDirection.UPSTREAM, 0, 1000, 2);
+    assertEquals(upstreamLineage.getTotal().intValue(), 0);
+    assertEquals(upstreamLineage.getRelationships().size(), 0);
+
+    EntityLineageResult downstreamLineage = service.getLineage(datasetOneUrn, LineageDirection.DOWNSTREAM, 0, 1000, 2);
+
+    assertEquals(downstreamLineage.getTotal().intValue(), 5);
+    assertEquals(downstreamLineage.getRelationships().size(), 5);
+    Map<Urn, LineageRelationship> relationships = downstreamLineage.getRelationships().stream().collect(Collectors.toMap(LineageRelationship::getEntity,
+            Function.identity()));
+    assertTrue(relationships.containsKey(datasetTwoUrn));
+    assertEquals(relationships.get(datasetTwoUrn).getDegree().intValue(), 1);
+    assertTrue(relationships.containsKey(datasetThreeUrn));
+    assertEquals(relationships.get(datasetThreeUrn).getDegree().intValue(), 2);
+    assertTrue(relationships.containsKey(datasetFourUrn));
+    assertEquals(relationships.get(datasetFourUrn).getDegree().intValue(), 2);
+    assertTrue(relationships.containsKey(dataJobOneUrn));
+    assertEquals(relationships.get(dataJobOneUrn).getDegree().intValue(), 1);
+    assertTrue(relationships.containsKey(dataJobTwoUrn));
+    assertEquals(relationships.get(dataJobTwoUrn).getDegree().intValue(), 1);
+
+    upstreamLineage = service.getLineage(datasetThreeUrn, LineageDirection.UPSTREAM, 0, 1000, 2);
+    assertEquals(upstreamLineage.getTotal().intValue(), 3);
+    assertEquals(upstreamLineage.getRelationships().size(), 3);
+    relationships = upstreamLineage.getRelationships().stream().collect(Collectors.toMap(LineageRelationship::getEntity,
+            Function.identity()));
+    assertTrue(relationships.containsKey(datasetOneUrn));
+    assertEquals(relationships.get(datasetOneUrn).getDegree().intValue(), 2);
+    assertTrue(relationships.containsKey(datasetTwoUrn));
+    assertEquals(relationships.get(datasetTwoUrn).getDegree().intValue(), 1);
+    assertTrue(relationships.containsKey(dataJobOneUrn));
+    assertEquals(relationships.get(dataJobOneUrn).getDegree().intValue(), 1);
+
+    downstreamLineage = service.getLineage(datasetThreeUrn, LineageDirection.DOWNSTREAM, 0, 1000, 2);
+    assertEquals(downstreamLineage.getTotal().intValue(), 0);
+    assertEquals(downstreamLineage.getRelationships().size(), 0);
+  }
```

Comment:
can you move this test into `GraphServiceTestBase.java`? This way we can make sure not to break both graph services as we make changes

Reply:
I have removed it from Neo4 tests and also From ElasticSearch tests.